### PR TITLE
Add unit test cases for store.go file

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
Added unit test cases for some functions of store.go file:

- `GetLinkByChannelID`
- `GetLinkByMSTeamsChannelID`
- `DeleteLinkByChannelID`
- `StoreChannelLink`
- `LinkPosts`
- `GetPostInfoByMSTeamsID`
- `GetPostInfoByMattermostID`
- `SetUserInfo`
- `TeamsToMattermostUserID`
- `MattermostToTeamsUserID`
- `GetTokenForMattermostUser`
- `GetTokenForMSTeamsUser`

#### Ticket Link
https://github.com/mattermost/mattermost-plugin-msteams-sync/issues/57

### CODE REVIEW

File: server/store/store_test.go [hunk 1]
```diff
@@ -1,18 +1,57 @@
...
+	db, container := createTestDB()
+	store.db = db
+	_ = store.Init()
+	return store, api, container
+}
+
+func createTestDB() (*sql.DB, testcontainers.Container) {
+	postgresPort := nat.Port("5432/tcp")
+	postgres, _ := testcontainers.GenericContainer(context.Background(),
+		testcontainers.GenericContainerRequest{
```
Comment (jespino) [inline server/store/store_test.go:36]: I love this!

File: server/store/store_test.go [hunk 2]
```diff
@@ -150,3 +189,228 @@ func TestCheckEnabledTeamByTeamID(t *testing.T) {
 		})
 	}
 }
+
+func TestStoreChannelLinkAndGetLinkByChannelID(t *testing.T) {
+	assert := assert.New(t)
+	store, api, container := setupTestStore(&plugintest.API{})
```
Comment (jespino) [inline server/store/store_test.go:195]:
```suggestion
	store, api, tearDownContainer := setupTestStore(&plugintest.API{})
	defer tearDownContainer()
```

File: server/store/store_test.go [hunk 3]
```diff
@@ -1,18 +1,57 @@
...
+					wait.ForLog("database system is ready to accept connections"),
+					wait.ForListeningPort(postgresPort),
+				),
+			},
+			Started: true,
+		})
+
+	hostPort, _ := postgres.MappedPort(context.Background(), postgresPort)
+	conn, _ := sqlx.Connect("postgres", fmt.Sprintf("postgres://user:pass@localhost:%s?sslmode=disable", hostPort.Port()))
+	return conn.DB, postgres
```
Comment (jespino) [inline server/store/store_test.go:54]:
```suggestion
	tearDownContainer := func() {
		if err := postgres.Terminate(ctx); err != nil {
			t.Fatalf("failed to terminate container: %s", err.Error())
		}
	}
	return conn.DB, tearDownContainer
```

File: server/store/store_test.go [hunk 4]
```diff
@@ -150,3 +189,228 @@ func TestCheckEnabledTeamByTeamID(t *testing.T) {
...
+	}
+
+	storeErr := store.StoreChannelLink(mockChannelLink)
+	assert.Nil(storeErr)
+
+	resp, getErr := store.GetLinkByChannelID("mockMattermostChannelID")
+	assert.Equal(mockChannelLink, resp)
+	assert.Nil(getErr)
+
+	termErr := container.Terminate(context.Background())
```
Comment (jespino) [inline server/store/store_test.go:217]: Of course if you follow the tear down approach you no longer need this two lines here

File: server/store/store_test.go [hunk 5]
```diff
@@ -150,3 +189,228 @@ func TestCheckEnabledTeamByTeamID(t *testing.T) {
...
+	assert.Nil(termErr)
+}
+
+func TestStoreChannelLinkdAndGetLinkByMSTeamsChannelID(t *testing.T) {
+	assert := assert.New(t)
+	store, api, container := setupTestStore(&plugintest.API{})
```
Comment (jespino) [inline server/store/store_test.go:223]: And the same everywhere else.

**Comment by ayusht2810** [comment]
@jespino fixed the review comments and updates some. Please re-review

File: server/store/store_test.go [hunk 6]
```diff
@@ -150,3 +233,210 @@ func TestCheckEnabledTeamByTeamID(t *testing.T) {
...
+
+	storeErr := store.SetUserInfo(testutils.GetID(), testutils.GetTeamUserID(), token)
+	assert.Nil(storeErr)
+
+	resp, getErr := store.GetTokenForMattermostUser(testutils.GetID())
+	assert.Equal(token, resp)
+	assert.Nil(getErr)
+}
+
+func TestSetUserInfoAndGetTokenForMSTeamsUser(t *testing.T) {
```
Comment (jespino) [inline server/store/store_test.go:423]: Most of these test cases need to cover the non-happy path. I would expect to test what happens here, for example, when I try to get a not existing MSTeamsUser.

**Review by jespino — CHANGES_REQUESTED** [review]
A couple of improvements needed, but is getting in very good shape.

**Comment by ayusht2810** [comment]
@jespino updated the test cases. Please re-review

**Review by jespino — APPROVED** [review]
now LGTM. Thanks!

**Pull request merged**